www/html/api-laravel/app/Repositories/Client/ClientRepositoryInterface.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace App\Repositories\Client;
+
+use App\Models\Client;
+
+interface ClientRepositoryInterface
+{
+    /**
+     * Insert new client into database
+     * @param array $data
+     * @return Client|null
+     */
+    public function create(array $data) : ?Client;
+
+    /**
+     * get one client by id into database
+     * @param int $id
+     * @return Client|null
+     */
+    public function getById(int $id) : ?Client;
+
+    /**
+     * Update one client by id into database
+     * @param int $id, array $data
+     * @return Client|null
```

Comment:
Sugestão utilizar uma linha para cada parametro

```suggestion
     * Update one client by id into database
     * @param int $id
     * @param array $data
     * @return Client|null
```